src/main/java/za/ac/cput/service/studentService/impl/StudentServiceImpl.java

```diff
@@ -0,0 +1,68 @@
+package za.ac.cput.service.studentService.impl;
+
+/*
+author: Ameer Ismail
+student nr: 218216033
+Service impl: StudentServiceImpl
+ADP 3 Assignment Group1
+Service implementation
+*/
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import za.ac.cput.entity.Student;
+import za.ac.cput.repository.IStudentRepository;
+import za.ac.cput.service.studentService.IStudentService;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class StudentServiceImpl implements IStudentService
+{
+    private final IStudentRepository repo;
+
+    @Autowired
+    public StudentServiceImpl(IStudentRepository repo) {
+        this.repo = repo;
+    }
+
+    // creating a student
+    @Override
+    public Student save(Student student)
+    {
+        return this.repo.save(student);
+    }
+
+    // reading a student
+    @Override
+    public Optional<Student> read(Long id)
+    {
+        return this.repo.findById(String.valueOf(id));
```

Comment:
You can pass the ID as is to the read method that's inherited from JPA. As long as you specified your Long type when you extended the JPA interface. It will accept a long type ID. There's no need to convert back and forth.

Remember JPA Interfaces and Classes are Generic so they can accept any primitive type.